x-pack/plugins/security_solution/public/detection_engine/rule_creation/components/related_integrations/related_integrations_help_info.tsx

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useState } from 'react';
+import { EuiLink, EuiPopover, EuiText, EuiButtonIcon } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { useKibana } from '../../../../common/lib/kibana';
+
+/**
+ * Theme doesn't expose width variables. Using provided size variables will require
+ * multiplying it by another magic constant.
+ *
+ * 320px width looks
+ * like a [commonly used width in EUI](https://github.com/search?q=repo%3Aelastic%2Feui%20320&type=code).
+ */
+const POPOVER_WIDTH = 320;
+
+export function RelatedIntegrationsHelpInfo(): JSX.Element {
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const { docLinks } = useKibana().services;
+
+  const onButtonClick = useCallback(() => setIsPopoverOpen((x) => !x), [setIsPopoverOpen]);
```

Comment:
nit: You may consider renaming `x` to `isOpen` to make it instantly readable

Reply:
Actually, since it's a common pattern, why not extract it into a reusable hook. WDYT?
```tsx
import { useToggle as _useToggle } from 'react-use';

export function useToggle(defaultValue: boolean): {
  value: boolean;
  toggle: () => void;
  setTrue: () => void;
  setFalse: () => void;
} {
  const [value, toggle] = _useToggle(defaultValue);

  const setTrue = useCallback(() => toggle(true), [toggle]);
  const setFalse = useCallback(() => toggle(false), [toggle]);

  return {
    value,
    toggle,
    setTrue,
    setFalse,
  };
}
```

```tsx
/* ... */
const { value: isPopoverOpen, toggle, setFalse: closePopover } = useToggle(false);

/* ... */
<EuiButtonIcon onClick={toggle} />
/* ... */
<EuiPopover button={button} isOpen={isPopoverOpen} closePopover={closePopover}>
```

Reply:
I think it's a good idea to use `useToggle` hook. It can be used directly and reduce the number of code lines.